### PR TITLE
Added combineVariants function

### PR DESCRIPTION
**Pull request by lfrancioli**
This function is needed to go from a split to a non-split representation, or to join non-split keytables by loci

### CODE REVIEW

File: src/main/scala/is/hail/expr/FunctionRegistry.scala [hunk 1]
```diff
@@ -777,6 +777,58 @@ object FunctionRegistry {
...
+      val (longer, shorter, swapped) = if (left.ref.length > right.ref.length) (left, right, false) else (right, left, true)
+      val ref_diff = longer.ref.substring(shorter.ref.length)
+
+      if (longer.ref.substring(0, shorter.ref.length) != shorter.ref)
+        fatal(s"Variants ref bases mismatch in combineVariants. Left ref: ${ left.ref }, right ref: ${ right.ref }")
+
+      val short_alleles_index = mutable.Map[Int, Int](0 -> 0)
+      val short_alleles = new mutable.ArrayBuffer[AltAllele](initialSize = shorter.nAltAlleles)
+
+      Range(0, shorter.nAltAlleles).foreach({
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: Prefer `(0 until shorter.nAltAlleles).foreach { `

File: src/main/scala/is/hail/expr/FunctionRegistry.scala [hunk 2]
```diff
@@ -777,6 +777,58 @@ object FunctionRegistry {
...
+            short_alleles += AltAllele(longer.ref, alt)
+            short_alleles_index(longer.nAltAlleles + short_alleles.length) = i + 1
+          }
+          else
+            short_alleles_index(new_index + 1) = i + 1
+      })
+
+      val newVariant = longer.copy(altAlleles = longer.altAlleles ++ short_alleles)
+      if (swapped)
+        Annotation(newVariant, short_alleles_index.toMap, Range(0, longer.nAltAlleles + 1).zipWithIndex.toMap)
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: `(0 to longer.nAltAlleles)`

Comment (tpoterba) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: `(0 to longer.nAltAlleles)`

File: src/main/scala/is/hail/expr/FunctionRegistry.scala [hunk 3]
```diff
@@ -777,6 +777,58 @@ object FunctionRegistry {
...
+      if (longer.ref.substring(0, shorter.ref.length) != shorter.ref)
+        fatal(s"Variants ref bases mismatch in combineVariants. Left ref: ${ left.ref }, right ref: ${ right.ref }")
+
+      val short_alleles_index = mutable.Map[Int, Int](0 -> 0)
+      val short_alleles = new mutable.ArrayBuffer[AltAllele](initialSize = shorter.nAltAlleles)
+
+      Range(0, shorter.nAltAlleles).foreach({
+        i =>
+          val alt = shorter.altAlleles(i).alt + ref_diff
+          val new_index = longer.altAlleles.indexWhere(_.alt == alt)
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: This is quadratic in complexity, would prefer to make a map from longer alt to index, then use `get` and match no that.

**Comment by lfrancioli** [comment]
Hey Tim, would be great if you could take a look :)

**Comment by lfrancioli** [comment]
Thanks for the review Tim! Didn't know about `x until y` and `x to y` constructs -- these are great!

**Pull request merged**